### DISCUSSION TITLE
Different grids issue when running a regression

### DISCUSSION THREAD

**Discussion by Murk89** [pydata/xarray#5840] [Q&A]
I am trying to run a regression with MOD13Q1 NDVI and my WRF climate model outputs. For this I need to assure that the MODIS raster and WRF maps are on the same projection and spatial resolution. So I transformed MODIS'S projection and resolution based on my WRF maps specs using Earth Engine and then converted the file to a netCDF. Using python's xarrays, I can see that the x and y dims of the MOD13Q1 dataset and corresponding WRF's south_north and west_east dims are quite different. Can someone advise on how to align these different datasets/maps in python to be able to run a regression at every grid box? Many thanks.

WRF map: (Time: 1, south_north: 111, west_east: 114)>
(![rcp45_snowdepthlessthan0 1](https://user-images.githubusercontent.com/59877746/136245792-52654c80-3638-487e-bba8-1b2b46e2e699.png)) 
![MOD13Q1](https://user-images.githubusercontent.com/59877746/136246020-f7b2b76a-2623-49a4-bc3d-c2350b70fb5c.png)
MOD13Q1: Dimensions:(x: 1378, y: 1203)

**Comment by andhuang-CLGX** [comment]
If they have both lat / lon coordinates:
xesmf
https://pangeo-xesmf.readthedocs.io/en/latest/notebooks/Rectilinear_grid.html

If not, reproject using rioxarray
https://corteva.github.io/rioxarray/stable/rioxarray.html?highlight=project#rioxarray.raster_array.RasterArray.reproject_match
cartopy.crs.transform_points or pyproj

then
interp_like
https://xarray.pydata.org/en/stable/generated/xarray.DataArray.interp_like.html

**Reply by Murk89** [reply]
Hi, 
Many thanks for the suggestions. 
I cannot seem to be able to use xesmf one because it is not supported by windows yet as stated on their installation guide page.

I am trying to use the reproject_match. And I get this error: `AttributeError: 'DataArray' object has no attribute 'rio'`

Though I have selected the relevant data arrays from my dataset, then even then used the reproject module. Any suggestions? 

My script is as follows: 
```python
import xarray as xr
import netCDF4
import h5netcdf
import rioxarray 
import pandas as pd

veg250m= xr.open_dataset("C:\\Desktop\mean_ndvi_250m.nc")
B1=veg250m.Band1

wrf=xr.open_dataset("C:\\Python files for plotting wrfoutput\\D02-2090-2100\\wrfout_d02_2090-11-01_00_00_00.nc")
temp=wrf.T2

trial=B1.rio.reproject_match(temp)
```

**Reply by keewis** [reply]
are you sure the `import rioxarray` worked correctly? If I'm not misunderstanding something, `rioxarray` is the package providing the `rio` accessor.

cc @snowman2

**Reply by Murk89** [reply]
I think it imported fine, because there isn't any error in importing it. 
is there a way to check if it is imported correctly?

**Reply by keewis** [reply]
maybe try accessing things like `rioxarray.open_rasterio` or print the version (`print(rioxarray.__version__)`)?

Edit: Something to be aware of is that if you have a file called `rioxarray.py` in any of the directories listed in `sys.path` (including the current directory) it might shadow the package (so the import succeeds but the code using it still fails).

**Reply by Murk89** [reply]
I think you are right.
Something is not working because there is this error as well: AttributeError: module 'rioxarray' has no attribute 'open_rasterio'
 I am comparatively new to using python, can you please suggest something in regards to deal with the shadowing of rioxarray then?

**Reply by keewis** [reply]
to figure out what file exactly it is that was imported, try to access `rioxarray.__file__`. It should return something like `.../lib/python3.x/site-packages/rioxarray/__init__.py`

**Reply by Murk89** [reply]
rioxarray.__file__  returns nothing.

**Reply by keewis** [reply]
that's strange. Can you look at your current directory to see if there's something like a `rioxarray.py` file? Or a directory called `rioxarray` with a `__init__.py` in it? If not, you will have to look into every directory listed in `sys.path` (use `import sys; print(sys.path)` to get those)

**Reply by Murk89** [reply]
So what I found is, `Myfolder\\envs\\gdal\\Lib\\site-packages\\rioxarray\\_pycache_\\_init_.cpython-38.pyc`

Also I am not sure if this has anything to do with it, but whenever I install packages through the anaconda prompt shell, it keeps saying:  set "GDAL_DRIVER_PATH=' 
gdal is the environment I have created where all libraries such as rioxarray and netcdf etc are installed.

**Reply by keewis** [reply]
hmm... I'd try with a new environment and see whether that fixes it.

**Reply by Murk89** [reply]
Ok thanks. I'll get back after trying it in a new environment.

**Reply by Murk89** [reply]
Hi Keewis, 
I was finally able to sort out the rioxarray, rasterio modules on my HPC rather than windows system laptop. 
Anyways, now I get this error: CRSError: Missing dst_crs. 
Would you know how to solve this please? Thanks. 

My code is:
#WRF datset 
wrf=xr.open_dataset("wrfout_d02_2007-12-01_00_00_00.nc", decode_coords="all" )
wrf = wrf.rename({'XTIME':'Time'})
wrf = wrf.assign_coords(Time=pd.to_datetime(wrf['Time'].values))

wrf.dims
#returns:
Dimensions:                (Time: 1, south_north: 111, west_east: 114)

#renaming the west_east and south_north dims of wrf to y and x. Because otherwise they are not recognized.
wrf = wrf.rename({'west_east':'y'})
wrf = wrf.rename({'south_north':'x'})


#MODIS dataset
ndvi=xr.open_dataset("mean_ndvi.nc", decode_coords="all")
ndvi['Band1'].rio.write_crs("epsg:4326", inplace=True)

ndvi.dims
#returns:

trial=ndvi.rio.reproject_match(wrf)

**Reply by Murk89** [reply]
Just to update
After trying to define the crs of the WRF dataset, which is explained here: https://spatialreference.org/ref/sr-org/wrf-lambert-conformal-conic/

The revised code is: 
#WRF datset
wrf=xr.open_dataset("wrfout_d02_2007-12-01_00_00_00.nc", decode_coords="all" )
wrf = wrf.rename({'XTIME':'Time'})
wrf = wrf.assign_coords(Time=pd.to_datetime(wrf['Time'].values))

wrf.dims
#returns:
Dimensions: (Time: 1, south_north: 111, west_east: 114)

#renaming the west_east and south_north dims of wrf to y and x. Because otherwise they are not recognized.
wrf = wrf.rename({'west_east':'y'})
wrf = wrf.rename({'south_north':'x'})

#writing the crs, using proj4
crs = CRS.from_string ('+proj=lcc +lat_1=64 +lat_2=15 +lat_0=64 +lon_0=10 +x_0=0 +y_0=0 +ellps=WGS84 +datum=WGS84 +units=m +no_defs')
wrf.T2.rio.write_crs("crs", inplace=True)

#MODIS dataset
ndvi=xr.open_dataset("mean_ndvi.nc", decode_coords="all")
ndvi.dims
#returns:
Frozen({'lat': 5949, 'lon': 12720})

#rename lat lon as y and x
ndvi=ndvi.rename({'lon':'x'})
ndvi=ndvi.rename({'lat':'y'})

#crs of the ndvi dataset
ndvi['Band1'].rio.write_crs("epsg:4326", inplace=True)

trial=ndvi.Band1.rio.reproject_match(wrf.T2)
trial.plot()

the above code produces the following plot:
![incorrectprojection](https://user-images.githubusercontent.com/59877746/137903632-a0bc6247-be50-4822-bf4e-6d5eb336d84b.png)


And I actually need it to look like this to be able to do a regression at each x,y 
![image](https://user-images.githubusercontent.com/59877746/137903793-11f34c70-e0cd-4d04-b788-54727a848388.png)


Can anyone please help in getting this right. I seem to be stuck.  Many thanks.

**Reply by snowman2** [reply]
You need to remove the string quotes around "crs" here: `wrf.T2.rio.write_crs("crs", inplace=True)`

**Reply by Murk89** [reply]
Hi, 

@snowman2 removing the string quotes generates this plot:
![without_stringquotes_crs](https://user-images.githubusercontent.com/59877746/138098347-4af32d06-28ff-465a-aaa0-5c2910163b88.png)

**Reply by snowman2** [reply]
Mind creating a discussion about this here: https://github.com/corteva/rioxarray/discussions ? Also, providing input datasets are very helpful for debugging.

**Reply by Murk89** [reply]
Yes why not. I'll create a new post. Hope this can be resolved.  I tried uploading the datasets, however .nc files were not supported. Is there a way around it? Thanks.
